### PR TITLE
Update Batch API version

### PR DESCRIPTION
**Pull request by jasper-schneider**
Pulled in the updated NuGet packages with the new API versions
Added a new cmdlet for the new REST API with tests + help
Re-recorded the scenario tests to use the new APIs (why the PR appears so big)

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@jasper-schneider__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (jaschnei). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/BatchAccountTests.ps1 [hunk 1]
```diff
@@ -46,6 +46,9 @@ function Test-CreatesNewBatchAccount
 		Assert-AreEqual $expected.Location $actual.Location
         Assert-AreEqual $expected.Tags[0]["Name"] $actual.Tags[0]["Name"]
 		Assert-AreEqual $expected.Tags[0]["Value"] $actual.Tags[0]["Value"]
+		Assert-True { $actual.CoreQuota -gt 0 }
+		Assert-True { $actual.PoolQuota -gt 0 }
+		Assert-True { $actual.ActiveJobAndJobScheduleQuota -gt 0 }
```
Comment (hovsepm) [inline src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/BatchAccountTests.ps1]: please align entire file. Replace tab with 4 spaces.

File: src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/Common.ps1 [hunk 2]
```diff
@@ -38,25 +38,29 @@ function Get-BatchAccountProviderLocation($index)
...
+		} 
+		else 
+		{  
+			if ($index -eq $null)
+			{
+				return "West US"
+			}
+			else
+			{
+				return $location[$index]
```
Comment (hovsepm) [inline src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/Common.ps1]: please align entire file. Replace tab with 4 spaces.

File: src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/JobTests.ps1 [hunk 3]
```diff
@@ -252,17 +252,17 @@ Tests querying for Batch jobs using a filter
...
-	$filter = "state eq'" + "$state" + "'"
+	$filter = "startswith(id,'$prefix')"
 
 	$jobs = Get-AzureBatchJob_ST -Filter $filter -BatchContext $context
 
 	Assert-AreEqual $matches $jobs.Length
 	foreach($job in $jobs)
 	{
-		Assert-True { $job.Id.StartsWith("$idPrefix") }
+		Assert-True { $job.Id.StartsWith("$prefix") }
```
Comment (hovsepm) [inline src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/JobTests.ps1]: please align entire file. Replace tab with 4 spaces.

File: src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/SubscriptionTests.ps1 [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+.SYNOPSIS
+Tests querying for subscription quotas in the Batch Service
+#>
+function Test-GetSubscriptionQuotas
+{
+	$location = Get-BatchAccountProviderLocation
+    $quotas = Get-AzureRmBatchSubscriptionQuotas $location
+
+	Assert-AreEqual $location $quotas.Location
+	Assert-True { $quotas.AccountQuota -gt 0 }
```
Comment (hovsepm) [inline src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/SubscriptionTests.ps1]: please align entire file. Replace tab with 4 spaces.

**Comment by markcowl** [comment]
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/254/

**Comment by jasper-schneider** [comment]
Looks like the Site Recovery tests failed in the on demand build.  I didn't touch that code, so I just hit the Retry button and will hope for the best - let me know if you'd like me to take any action.
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/261/

**Comment by markcowl** [comment]
@jasper-schneider you will most likely need to pull the latest changes from the dev branch at any rate

**Comment by markcowl** [comment]
@jasper-schneider Updated run is here:  http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/265/  I believe this was a machine-specific issue.  But this run will likely fail on the SQL tests unless you sync with the latest

**Comment by jasper-schneider** [comment]
Ok, just did a merge of upstream/dev

**Comment by markcowl** [comment]
@jasper-Schneider That should do it.  New on-demand run will he here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/268/

**Pull request merged**